contrib/sonatypecentral/src/mill/contrib/sonatypecentral/SonatypeCentralPublishModule.scala

```diff
@@ -0,0 +1,143 @@
+package mill.contrib.sonatypecentral
+
+import com.lumidion.sonatype.central.client.core.{PublishingType, SonatypeCredentials}
+import mill._
+import scalalib._
+import define.{ExternalModule, Task}
+import mill.api.Result
+import mill.contrib.sonatypecentral.SonatypeCentralPublishModule.{
+  defaultAwaitTimeout,
+  defaultConnectTimeout,
+  defaultCredentials,
+  defaultReadTimeout,
+  getPublishingTypeFromReleaseFlag,
+  getSonatypeCredentials
+}
+import mill.scalalib.PublishModule.{defaultStringGpgArgs, getFinalGpgArgs}
+import mill.scalalib.publish.Artifact
+import mill.scalalib.publish.SonatypeHelpers.{
+  PASSWORD_ENV_VARIABLE_NAME,
+  USERNAME_ENV_VARIABLE_NAME
+}
+
+trait SonatypeCentralPublishModule extends PublishModule {
+  def gpgArgs: T[String] = T { defaultStringGpgArgs }
+
+  def connectTimeout: T[Int] = T { defaultConnectTimeout }
+
+  def readTimeout: T[Int] = T { defaultReadTimeout }
+
+  def awaitTimeout: T[Int] = T { defaultAwaitTimeout }
+
+  def shouldRelease: T[Boolean] = T { true }
+
+  def publishSonatypeCentral(
+      username: String = defaultCredentials,
+      password: String = defaultCredentials
+  ): define.Command[Unit] =
+    T.command {
+      val publishData = publishArtifacts()
+      val fileMapping = publishData.toDataWithConcretePath._1
+      val artifact = publishData.meta
+      val finalCredentials = getSonatypeCredentials(username, password)()
+
+      val publisher = new SonatypeCentralPublisher(
+        credentials = finalCredentials,
+        gpgArgs = getFinalGpgArgs(gpgArgs()),
+        connectTimeout = connectTimeout(),
+        readTimeout = readTimeout(),
+        log = T.log,
+        workspace = T.workspace,
+        env = T.env,
+        awaitTimeout = awaitTimeout()
+      )
+      publisher.publish(fileMapping, artifact, getPublishingTypeFromReleaseFlag(shouldRelease()))
+    }
+}
+
+object SonatypeCentralPublishModule extends ExternalModule {
+
+  private val defaultCredentials = ""
+  private val defaultReadTimeout = 60000
+  private val defaultConnectTimeout = 5000
+  private val defaultAwaitTimeout = 120 * 1000
+  private val defaultShouldRelease = true
```

Comment:
Defaults should be `public` so API users can access them.

Reply:
Done.